bib/internal/uploader/aws.go

```diff
@@ -12,19 +12,51 @@ import (
 	"github.com/osbuild/images/pkg/cloud/awscloud"
 )
 
+func checkAWSCredentialFileExists() (bool, error) {
+	// SMELL make this path configurable?
+	credentialFile := "$HOME/.aws/credentials"
+
+	expandedFilename := os.ExpandEnv(credentialFile)
+	_, err := os.Stat(expandedFilename)
+	if err != nil {
+		return false, fmt.Errorf("AWS credential file not found: %s", expandedFilename)
+	}
+	return true, nil
+}
+
 // NewAWSClient returns an awscloud.AWS configured with a session for a given
 // region. It reads the credentials from environment variables:
 // AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY.
-func NewAWSClient(region string) (*awscloud.AWS, error) {
+// or tries to use "$HOME/.aws/credentials" otherwise
+func NewAWSClient(region string, awsProfile string) (*awscloud.AWS, error) {
```

Comment:
I find the current function a bit entangled and wonder if this could be made a bit easier to read (right now it's a bit hard for me to see what takes precendence for example). My gut feeling is that it might be easier if we would extract two helpers from the function and write this as, maybe something like this:
```go
func newAWSClientFromEnv(region string, awsProfile string) (*awscloud.AWS, error) {
	keyID := os.Getenv("AWS_ACCESS_KEY_ID")
	if keyID == "" {
		return nil, fmt.Errorf("AWS_ACCESS_KEY_ID must be set")
	}
	secretKey := os.Getenv("AWS_SECRET_ACCESS_KEY")
	if secretKey == "" {
		return nil, fmt.Errorf("AWS_SECRET_ACCESS_KEY must be set")
	}
	return awscloud.New(region, keyID, secretKey, "")
}

func newAWSClientFromCredsFile(region string, awsProfile string) (*awscloud.AWS, error) {
	foundCredentialFile, err := checkAWSCredentialFileExists()
	if err != nil {
		return nil, err
	}
	if !foundCredentialFile {
		return nil, fmt.Errorf("cannot find credentials file %v", credentialFile)
	}
	if awsProfile != "default" {
		os.Setenv("AWS_PROFILE", awsProfile)
	}
	return awscloud.NewDefault(region)
}

// NewAWSClient returns an awscloud.AWS configured with a session for a given
// region. It reads the credentials from environment variables:
// AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY.
// or tries to use "$HOME/.aws/credentials" otherwise
func NewAWSClient(region string, awsProfile string) (*awscloud.AWS, error) {
	client, errEnv := newAWSClientFromEnv(region, awsProfile)
	if errEnv == nil {
		return client, nil
	}
	client, errCredsFile := newAWSClientFromCredsFile(region, awsProfile)
	if errCredsFile == nil {
		return client, nil
	}

	return nil, fmt.Errorf("cannot usw AWS, neither env nor file method worked\n- env: %v\n- file: %v", errEnv, errCredsFile)
}
```
this way it's more linear and it's clear which of the two takes precedence. This assumes of course that I understand the existing flow which I might not have :) (if so, sorry! it's still early here).

And I would absolutely love to see a test for this too :)

Reply:
I wonder if we cannot just use `awscloud.NewDefault()`. According to [the docs](https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-credentials), it checks the env variables first, then it checks the credentials file, and then it even checks some more places (if IAM roles also work out of the box, that would be sweat).

Reply:
You are right - the main question for me is if we should provide commandline arguments and translate them to os.Setenv for convenience - as the approach with the env file does not seem to be very convenient (for all usecases)
I can rewrite this and check the compatibility for sure

Reply:
For `AWS_ACCESS_KEY_ID` and the other one: They basically have to be `--env-file`, you don't want to pass secrets as `argv`.

For the credentials file: I guess the options are:
```
podman run \
  -v $HOME/.aws:/root/.aws:ro \
  quay.io/centos-bootc/bootc-image-builder:latest \
  --aws-profile default
```

vs.

```
podman run \
  -v $HOME/.aws:/root/.aws:ro \
  --env AWS_PROFILE=default
  quay.io/centos-bootc/bootc-image-builder:latest
```

(you don't have to use `--env-file`, you can just `--env` for one env variable)

I think I might actually prefer the latter one, because the options are closely together. Not a strong opinion, though. @mvo5, wdyt?